test/RelayPricer.test.ts

```diff
@@ -35,69 +35,82 @@ describe('RelayPricer', () => {
         });
 
         it('should initiate sourceApi', async () => {
-            const coinBase = new CoinBase();
-            const pricer = new RelayPricer(coinBase);
+            const pricer = new RelayPricer(fakeSourceApi);
             expect(pricer).to.be.instanceOf(RelayPricer);
-            expect(pricer).to.have.property('sourceApi', coinBase);
+            expect(pricer).to.have.property('sourceApi', fakeSourceApi);
         });
 
-        it('should return source currency conversion price', async () => {
-            buildFakePrices(FAKE_PRICE_1, FAKE_PRICE_2);
-            const price = await pricer.getPrice(
-                SOURCE_CURRENCY,
-                TARGET_CURRENCY
+        it('should fail if sourceApi is null', async () => {
+            expect(() => new RelayPricer(null)).to.throw(
```

Comment:
```suggestion
            expect(() => new RelayPricer(undefined as unknown as SourceApi)).to.throw(
```
I'm surprised TS didn't complain about it